File: collection.scala
import scala.util.Random

def repeat(n: Int)(f: => Unit) {
  if (n > 0) { f; repeat(n-1)(f) }
}

val start = System.currentTimeMillis
repeat(100) {
  val alphabet = ('a' to 'z').toArray
  val letterList = (0 to 100000).map( _ => alphabet(Random.nextInt(26)) )
  val pick = letterList.distinct(Random.nextInt(26))
  val (before, after) = letterList.partition{ _ > pick }
}
println(System.currentTimeMillis - start)

File: collection.rb
require "benchmark"

ALPHABET = ('a'..'z').to_a

def iterate(count)
  count.times do
    letter_list = []
    100_000.times { letter_list << ALPHABET[rand(26)] }
    pick = letter_list.uniq[rand(26)]
    before, after = letter_list.partition { |n| n > pick }
  end
end

Benchmark::bmbm do |x|
  x.report("10000") do
    iterate(100)
  end

  x.report("100") do
    iterate(100)
  end
end